std/conv.d

```diff
@@ -1883,6 +1883,47 @@ if (isInputRange!S && !isInfinite!S && isSomeChar!(ElementEncodingType!S) &&
     assert(to!int(byDchar("10"), 10) == 10);
 }
 
+/**
+String, or string-like input range, to char type not directly
+supported by parse parses the first dchar of the source.
+
+Returns: the first code point of the input range, converted
+         to type T.
+
+Throws: ConvException if the input range contains more than
+        a single code point, or if the code point does not
+        fit into a code unit of type T.
+*/
+private T toImpl(T, S)(S value)
+if (isSomeChar!T && !is(typeof(parse!T(value))) &&
+    is(typeof(parse!double(value))))
+{
+    immutable dchar codepoint = parse!dchar(value);
+    if (!value.empty)
+        throw new ConvException(convFormat("Cannot convert \"%s\" to %s because it " ~
+                                           "contains more than a single code point.",
+                                           value, T.stringof));
+    auto equivalentTArray = to!(T[])(codepoint);
```

Comment:
Use `std.utf.encode`, as `to!(T[])` will allocate.

```D
T[dchar.sizeof / T.sizeof] outbuf; // hacky sizing, but it's correct.
auto resultSize = encode(outbuf, codepoint);
if(resultSize != 1) { ... } // throw conv error.
return outbuf[0];
```

Reply:
[As per the documentation](https://dlang.org/phobos/std_utf.html#encode) encode will throw UTFException if the input is not a valid UTF code point. Should I catch said exception and rethrow a ConvException?

Reply:
Let's follow precedent -- whatever `to!wstring(string)` does with an invalid code point. I suspect it lets the exception through.

Reply:
Okay, no further changes then

Reply:
Yep, just tested, throws a `UTFException`.